### ISSUE TITLE
systemd codedeploy-agent.service installed with executable permissions (systemd-sysv-generator)

### ISSUE THREAD

**Issue by MarkSonghurst** [aws/aws-codedeploy-agent#205]
CentOS 7.latest
I've noticed that within the systemd service directory `/usr/lib/systemd/system/` that the `codedeploy-agent.service file` is being installed with executable permissions. This is the only .service file I have which has the executable permission set. A brief example:

```
-rw-r--r--. 1 root root  491 Nov 16 19:17 cloud-config.service
-rw-r--r--. 1 root root  536 Nov 16 19:17 cloud-config.target
-rw-r--r--. 1 root root  514 Nov 16 19:17 cloud-final.service
-rw-r--r--. 1 root root  878 Nov 16 19:17 cloud-init-local.service
-rw-r--r--. 1 root root  648 Nov 16 19:17 cloud-init.service
-rwxr--r--. 1 root root  460 Oct 22 19:55 codedeploy-agent.service
-rw-r--r--. 1 root root  787 Jan 29 17:56 console-getty.service
-rw-r--r--. 1 root root  749 Jan 29 17:56 console-shell.service
-rw-r--r--. 1 root root  808 Jan 29 17:56 container-getty@.service
-rw-r--r--. 1 root root  294 Feb  1 15:04 cpupower.service
-rw-r--r--. 1 root root  284 Nov 20 14:22 crond.service
```

Besides my OCD ;-) this is causing an Alert to be logged in our monitoring system:
```
Identifier: systemd-sysv-generator 
Message: Configuration file /usr/lib/systemd/system/codedeploy-agent.service is marked executable. Please remove executable permission bits. Proceeding anyway.
```

Here are some more details:
```
CODE_FILE: src/shared/util.c
CODE_FUNCTION: fd_warn_permissions
CODE_LINE: 7132
SYSLOG_FACILITY: 3
_BOOT_ID: XXXX
_CAP_EFFECTIVE: 1fffffffff
_CMDLINE: /usr/lib/systemd/systemd --system --deserialize 24
_COMM: systemd
_EXE: /usr/lib/systemd/systemd
_GID: 0
_HOSTNAME: XXXX
_MACHINE_ID: XXXX
_SELINUX_CONTEXT: system_u:system_r:init_t:s0
```

It would be great if the permission could be installed at a suitable level, unless executable is necessary as some part of the install process?
Thank You.

**Comment by edwinchana** [comment]
+1, want to know the answer.

**Comment by hsnopi** [comment]
+1

**Comment by annamataws** [comment]
Hi,

We will look into this and get back to you. 
Thank you

**Comment by Tomicapo** [comment]
Hi. Same here with Amazon Linux 2.

Error message:

> systemd-sysv-generator[4086]: Configuration file /usr/lib/systemd/system/codedeploy-agent.service is marked executable. Please remove executable permission bits. Procesding anyway.

**Comment by tarmish** [comment]
Having the same issue with CentOS 7 - started 28th Feb 2019.

**Comment by elpy1** [comment]
Any update on when we can expect this to be corrected? It appears the rpm needs to be rebuilt.

For anyone that wants to check, you can query the rpm that's installed by the installer script `codedeploy_install` (https://aws-codedeploy-ap-southeast-2.s3.amazonaws.com/latest/install):

Dump the list of files and see the mode/permissions is `0744`:
```
[root@test ~]# rpm --query --package --dump  codedeploy-agent-1.1.0-4.noarch.rpm | grep '\.service'
/usr/lib/systemd/system/codedeploy-agent.service 460 1591663560 bb1922c6dbfe85007398e98e92e8d883 0100744 root root 0 0 0 X
```

You can also test/confirm this by changing the file's permissions, and then using `rpm` to restore them to original state:
```
[root@test ~]# ls -l /usr/lib/systemd/system/codedeploy-agent.service 
-rwxr--r--. 1 root root 460 Jun  9 10:46 /usr/lib/systemd/system/codedeploy-agent.service

[root@test ~]# chmod 644 /usr/lib/systemd/system/codedeploy-agent.service
[root@test ~]# ls -l /usr/lib/systemd/system/codedeploy-agent.service 
-rw-r--r--. 1 root root 460 Jun  9 10:46 /usr/lib/systemd/system/codedeploy-agent.service

[root@test ~]# rpm --restore codedeploy-agent

[root@test ~]# ls -l /usr/lib/systemd/system/codedeploy-agent.service 
-rwxr--r--. 1 root root 460 Jun  9 10:46 /usr/lib/systemd/system/codedeploy-agent.service
```

**Comment by puneethnr** [comment]
This is fixed in versions 1.4.0 and after. Recommend installing the latest version of the agent.
